libcalico-go/lib/validator/v3/validator.go

```diff
@@ -1673,6 +1673,16 @@ func validateBGPConfigurationSpec(structLevel validator.StructLevel) {
 			}
 		}
 	}
+
+	// Check that node mesh password cannot be set if node to node mesh is disabled.
+	if spec.NodeMeshPassword != nil && spec.NodeToNodeMeshEnabled != nil && !*spec.NodeToNodeMeshEnabled {
```

Comment:
Do we need an explicit check that this is only set on the deafult BGP config? I think probably not, because node-mesh can only be enabled on the default?

Reply:
Yeah, there's already a check in the client that prevents `NodeToNodeMeshEnabled` from being set on any BGP config other than the default config. Can we circumnavigate those checks by going straight through the K8s API?